gitops/generate_build_test.go

```diff
@@ -184,10 +185,59 @@ func TestGenerateInitialBuildPipelineRun(t *testing.T) {
 				},
 			},
 		},
+		{
+			name:         "generate initial build pipelien run no pull secret",
+			noPullSecret: true,
+			args: args{
+				component: component,
+			},
+			want: tektonapi.PipelineRun{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "testcomponent-",
+					Namespace:    "kcpworkspacename",
+					Labels:       getBuildCommonLabelsForComponent(&component),
+				},
+				Spec: tektonapi.PipelineRunSpec{
+					PipelineRef: &tektonapi.PipelineRef{
+						Bundle: gitopsConfig.BuildBundle,
+						Name:   "noop",
+					},
+					Params: []tektonapi.Param{
+						{
+							Name: "git-url",
+							Value: tektonapi.ArrayOrString{
+								Type:      tektonapi.ParamTypeString,
+								StringVal: "https://host/git-repo",
+							},
+						},
+						{
+							Name: "output-image",
+							Value: tektonapi.ArrayOrString{
+								Type:      tektonapi.ParamTypeString,
+								StringVal: "",
+							},
+						},
+					},
+					Workspaces: []tektonapi.WorkspaceBinding{
+						{
+							Name: "workspace",
+							PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+								ClaimName: "appstudio",
+							},
+							SubPath: "testcomponent/" + getInitialBuildWorkspaceSubpath(),
+						},
+					},
+				},
+			},
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			if got := GenerateInitialBuildPipelineRun(tt.args.component, gitopsConfig); !reflect.DeepEqual(got, tt.want) {
+			testGitOptsConfig := gitopsConfig
+			if tt.noPullSecret {
+				testGitOptsConfig = prepare.GitopsConfig{BuildBundle: "quay.io/redhat-appstudio/build-templates-bundle:0.0.1"}
```

Comment:
well if I understand what you are suggesting, then don't we a conditional around the call to `GenerateInitialBuildPipelineRun` to vary which gitopsConfig we use, right?

Or maybe I'm not quite following you?

Can you post a code snippet to clarify @brunoapimentel ?

thanks

Reply:
```
gitOpsConfig = prepare.GitopsConfig{BuildBundle: buildBundle, AppStudioPullSecretPresent: ! tt.noPullSecret}

if got := GenerateInitialBuildPipelineRun(tt.args.component, gitOpsConfig); !reflect.DeepEqual(got, tt.want) {
        t.Errorf("GenerateInitialBuildPipelineRun() = %v, want %v", got, tt.want)
}
```
And change L127 to simply:
```
buildBundle := "quay.io/redhat-appstudio/build-templates-bundle:0.0.1"
```
(that being said, I'd also change the `tt.noPullSecret` to `tt.PullSecretPresent` to avoid a double negative :slightly_smiling_face:  )

Reply:
ahhh ... ok thanks @brunoapimentel yeah I like that better 

will update shortly

Reply:
on the `noPullSecret` name, I went with that `no...` because I wanted to original behavior to be the "default" and hence the field's defaulting to false left that test instance untouched.

that said, there are only 2 test instances right now, so it is 50 / 50 either way.

How about I go with `pullSecretMissing` and then I can keep the same defaulting I had before?